certbot-ci/certbot_integration_tests/apache_tests/apache_config.py

```diff
@@ -0,0 +1,171 @@
+import shutil
+import subprocess
+import os
+
+import pkg_resources
+import getpass
+
+
+def construct_apache_config_dir(apache_root, http_port, https_port, key_path=None,
+                                cert_path=None, wtf_prefix='le'):
+    config_path = os.path.join(apache_root, 'config')
+    shutil.copytree('/etc/apache2', config_path, symlinks=True)
```

Comment:
This makes the behavior of these tests reliant on the machine they are run on and things like the directories included in the main config file below are Debian specific. I think in an ideal world, we'd remove all this and make the Apache configuration independent of the machine it is run on like is done in the nginx integration tests, but I think doing so is hard/impossible because Apache binaries normally need things like shared object files to enable TLS.

Since this is not feasible and [configuring the Apache plugin to use an alternate directory is not fully supported from inside the plugin](https://github.com/certbot/certbot/issues/7385) and as a result requires a lot of extra complexity in the tests, I think we should run these tests in Docker where we have a consistent environment and just modify `/etc/apache2` as we wish.

Stepping back even further about how to test the Apache plugin though, I think one of the most complicated and necessary to test parts of the plugin is all the overrides on different OSes. I think it'd be nice if in the long term the Apache integration tests and the "test farm tests" that do this shared as much code as possible. I don't see why there needs to be any difference between the Apache integration tests run on one system and the tests run as part of the future "test farm" tests on the same system.

With all this in mind, I think it may make sense for us to block adding Apache integration tests on rewriting the test farm tests. Then as part of testing PRs or snaps, we could run the Apache "test farm tests" but on a smaller set of systems to speed things up. There are other ways we could go about reusing code between these two types of tests too, but this to me initially seems like it might be the simplest.

What do you think?